oscarapi/views/basket.py

```diff
@@ -126,7 +127,11 @@ def add_voucher(request, format=None):
                                              context={'request': request})
     if v_ser.is_valid():
         basket = get_basket(request)
-        voucher = v_ser.object
```

Comment:
This is a n API change how this works in the restframework, see: http://www.django-rest-framework.org/topics/3.0-announcement/#serializers and then especially:

```Validating the data would create an object instance. This instance would be available as serializer.object.```

We used this mechanism to retrieve the object itself, even if it was not saved (for example in the login serializer). However, with the add-voucher call we don't want the object to be created or upated (which is now supported in the .create and .update methods of a serializer) as we just want to check if the voucher exists and if it's valid.

So we could set the .instance property in the serializer to the validated correct voucher (self.instance = voucher), or do it like this.